security/authorization/compiler/api/compiler.go

```diff
@@ -0,0 +1,145 @@
+// +build go1.10
+
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package compiler is a utility library containing functions to compile
+// a User policy yaml into a serialized RBAC proto object
+package compiler
+
+import (
+	"io/ioutil"
+	"log"
+
+	pb "github.com/envoyproxy/go-control-plane/envoy/config/rbac/v2" // v3
+	cel "github.com/google/cel-go/cel"
+	decls "github.com/google/cel-go/checker/decls"
+	expr "google.golang.org/genproto/googleapis/api/expr/v1alpha1"
+	"google.golang.org/protobuf/proto"
+	"gopkg.in/yaml.v2"
+)
+
+// UserPolicy is the user policy
+type UserPolicy struct {
+	Action string `yaml:"action"`
+	Rules  []struct {
```

Comment:
The format seems not right? Did you run `go fmt` before submitting?

Reply:
Sorry I forgot to mention that in order for your tests to be run, you have to change Makefile. See https://github.com/grpc/grpc-go/pull/3707/files for how to make changes.

Reply:
added to the makefile and I double checked with go fmt and nothing changed